### DISCUSSION TITLE
[supa_audit] Get rows by record id value, or how to query record_id uuid

### DISCUSSION THREAD

**Discussion by kavitang** [supabase/supabase#13858] [Questions]
Hey,

I'm trying to query the `record_version` table in the `audit` schema by the value of `record --> id` field. This is obviously a very heavy query as this is a JSON column and the table is very heavy. The `record_id` field is a combination of entity_oid, and primary key value, but how can I query by this field?
Should I generate the uuid using the same function `to_record_id`? I'm unclear on how I can query this table when I have the `id` value for a table and want to fetch all logs related to this record id. 

Any suggestions on how this can be achieved would be highly appreciated. Thanks!

**Comment by tatupesonen** [comment]
Also interested!

**Comment by tatupesonen** [comment]
@kavitang I figured out a solution. If you can manually build the query, you can execute something like this:
```sql
select audit.to_record_id('public.your_table'::regclass::oid, audit.primary_key_columns('public.your_table'::regclass::oid)::text[], '{"your": "json goes here"}' );
```

That will give you back the record_id that you can then use in your queries.
